test/operators/test_block_diag_linear_operator.py

```diff
@@ -67,6 +67,36 @@ def evaluate_linear_op(self, linear_op):
                     actual[i, k, j * 4 : (j + 1) * 4, j * 4 : (j + 1) * 4] = blocks[i, k, j]
         return actual
 
+class TestBlockDiagLinearOperatorMetaClass(unittest.TestCase):
+    def test_metaclass_constructor(self):
+        k, n = 3, 5 # number of blocks, block size
+        b1, b2 = 2, 3 # batch dimensions
+        base_operators = [torch.randn(k, n), torch.randn(b1, b2, k, n)]
+
+        for base_op in base_operators: # repeats tests for both batched and non-batched tensors
+            base_diag = DiagLinearOperator(base_op)
```

Comment:
could use [`subTest`](https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests) here to make the test failures easier to parse